pkg/sdn/plugin/eventqueue.go

```diff
@@ -0,0 +1,140 @@
+package plugin
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/client/cache"
+)
+
+func NewEventQueue(keyFunc cache.KeyFunc) *cache.DeltaFIFO {
+	return cache.NewDeltaFIFO(
+		keyFunc,
+		cache.DeltaCompressorFunc(func(d cache.Deltas) cache.Deltas {
+			return deltaCompressor(d, keyFunc)
+		}),
+		cache.NewStore(keyFunc),
```

Comment:
This doesn't seem like it should work; the last argument to NewDeltaFIFO() is treated read-only by DeltaFIFO, so by passing an empty Store (which will always remain an empty Store) you are effectively claiming that you never know about any objects, even after being told about them. This should cause you to lose Delete notifications.